bluesky/plan_stubs.py

```diff
@@ -873,3 +875,78 @@ def move():
     motors = step.keys()
     yield from move()
     yield from trigger_and_read(list(detectors) + list(motors))
+
+
+def repeat(plan, num=1, delay=None):
+    """
+    Repeat a plan num times with delay between each repeat.
+
+    Parameters
+    ----------
+    plan: callable or list
+        Callable that returns a valid plan, or a list of Message objects
+    num : integer, optional
+        number of readings to take; default is 1
+
+        If None, capture data until canceled
+    delay : iterable or scalar, optional
+        time delay between successive readings; default is 0
+
+    Notes
+    -----
+    If ``delay`` is an iterable, it must have at least ``num - 1`` entries or
+    the plan will raise a ``ValueError`` during iteration.
+    """
+    # If delay is a scalar, repeat it forever. If it is an iterable, leave it.
+    if not isinstance(delay, Iterable):
+        delay = itertools.repeat(delay)
+    else:
+        try:
+            num_delays = len(delay)
+        except TypeError:
+            # No way to tell in advance if we have enough delays.
+            pass
+        else:
+            if num - 1 > num_delays:
+                raise ValueError("num=%r but delays only provides %r "
+                                 "entries" % (num, num_delays))
+        delay = iter(delay)
+
+    def finite_plan():
+        for i in range(num):
+            now = time.time()  # Intercept the flow in its earliest moment.
+            if isinstance(plan, list):
```

Comment:
Use `ensure_generator` from `bluesky.utils` which cases regularizes this and other possibilities.